pymongo/common.py

```diff
@@ -524,56 +525,78 @@ def validate_tzinfo(dummy, value):
     return value
 
 
-# journal is an alias for j,
-# wtimeoutms is an alias for wtimeout,
-URI_VALIDATORS = {
-    'replicaset': validate_string_or_none,
-    'w': validate_non_negative_int_or_basestring,
-    'wtimeout': validate_non_negative_integer,
-    'wtimeoutms': validate_non_negative_integer,
-    'fsync': validate_boolean_or_string,
-    'j': validate_boolean_or_string,
+# Dictionary where keys are the names of public URI options, and values
+# are lists of aliases for that option. Aliases of option names are assumed
+# to have been deprecated.
+URI_OPTIONS_ALIAS_MAP = {
+    'journal': ['j'],
+    'wtimeoutms': ['wtimeout'],
+    'tls': ['ssl'],
+    'tlsallowinvalidcertificates': ['ssl_cert_reqs'],
+    'tlsallowinvalidhostnames': ['ssl_match_hostname'],
```

Comment:
The same problem exists here. tlsallowinvalidhostname=true should be equivalent to ssl_match_hostname=False.

Reply:
In this case you are correct - I didn't account for the inversion. I have done so in my latest commits by using a lambda to invert the return value of the validator function.